R/coh_data_wrangling.R

```diff
@@ -168,27 +167,35 @@ get_immunization_date <- function(data,
     vacc_date_col,
     min.len = 1L
   )
+
   # check data has expected column names
   checkmate::assert_names(
     names(data),
     must.include = c(outcome_date_col, vacc_date_col)
   )
-  # TODO: check that outcome, vax, and end date columns are of type Date
-  # or coercible to date
-
-  # check outcome and immunization delay as a number - may be decimal for now
-  # TODO: consider restricting to a count or integerish
-  checkmate::assert_number(
-    outcome_delay,
-    lower = 0, finite = TRUE
+
+  # check if date columns are date type
+  checkmate::assert_date(
+    data[[outcome_date_col]]
   )
-  checkmate::assert_number(
-    immunization_delay,
-    lower = 0, finite = TRUE
+  for (i in seq_along(vacc_date_col)) {
+    checkmate::assert_date(
+      data[[vacc_date_col[i]]]
+    )
+  }
+
+  # check outcome and immunization delay
+  stopifnot(
+    "Please provide a non-null integer number greater or equal than 0
+    in `outcome_delay`. Use round(`outcome_delay`,0)" =
+      checkmate::test_integerish(outcome_delay, lower = 0, null.ok = FALSE),
+    "Please provide a non-null integer number greater or equal than 0
+    in `outcome_delay`. Use round(`immunization_delay`,0)" =
+      checkmate::test_integerish(immunization_delay, lower = 0, null.ok = FALSE)
   )
 
-  # expect end cohort is a string - will be coerced to date later
-  checkmate::assert_string(
+  # expect end cohort is a date
+  checkmate::assert_date(
     end_cohort
   )
 
```

Comment:
If I understand correctly the end of the study is given by a single date, in which case it would be good to check for a single, non-missing value.
```suggestion
  checkmate::assert_date(
    end_cohort, any.missing = FALSE, len = 1
  )

```

Reply:
Right, thanks for the suggestion.